spec.bs

```diff
@@ -6,57 +6,882 @@ Status: CG-DRAFT
-=============================================
 <xmp class='idl'>
+  [Exposed=(Window)]
+  interface SharedStorageWorklet : Worklet {
+  };
+</xmp>
+
+  ## The {{SharedStorageWorkletGlobalScope}} ## {#global-scope}
+
+  The {{SharedStorageWorklet}}'s [=worklet global scope type=] is {{SharedStorageWorkletGlobalScope}}. 
+
+  <xmp class='idl'>
+    [Exposed=SharedStorageWorklet, Global=SharedStorageWorklet]
+    interface SharedStorageWorkletGlobalScope : WorkletGlobalScope {
+      undefined register(DOMString name,
+                         SharedStorageOperationConstructor operationCtor);
+    };
+
+    callback SharedStorageOperationConstructor = SharedStorageOperation(object options);
+
+    [Exposed=SharedStorageWorklet]
+    interface SharedStorageOperation {
+    };
+  </xmp>
+
+  Each {{SharedStorageWorkletGlobalScope}} has an associated [=environment settings object=] <dfn for=SharedStorageWorkletGlobalScope>outside settings</dfn>, which is the associated {{SharedStorageWorklet}}'s [=relevant settings object=].
+
+  The {{SharedStorageWorkletGlobalScope}}'s [=module map=]'s [=module scripts=] should each define and {{register}} one or more {{SharedStorageOperation}}s. 
+
+  Each {{SharedStorageWorkletGlobalScope}} also has an associated <dfn for=SharedStorageWorkletGlobalScope>operation map</dfn>, which is an [=map=], initially empty, of [=strings=] (denoting operation names) to [=functions=].
+
+  Currently each {{SharedStorageOperation}} registered via {{SharedStorageWorkletGlobalScope/register()}} must be one of the following two types:
+  * {{SharedStorageRunOperation}}
+  * {{SharedStorageSelectURLOperation}}
+
+  A {{SharedStorageRunOperation}} is an {{SharedStorageOperation}} that has no JavaScript return value, and hence is especially suited for calling into the [=private aggregation=] service.
+
+  A {{SharedStorageSelectURLOperation}} is an {{SharedStorageOperation}} that takes in a [=list=] of {{SharedStorageUrlWithMetadata}}s (i.e. [=dictionaries=] containing [=strings=] representing {{URL}}s each wrapped with metadata) and returns a {{long}} integer index specifying which of these {{URL}}s should be selected.
+
+  <xmp class='idl'>
+    [Exposed=SharedStorageWorklet]
+    interface SharedStorageRunOperation : SharedStorageOperation {
+      Promise<undefined> run(object data);
+    };
+
+    [Exposed=SharedStorageWorklet]
+    interface SharedStorageSelectURLOperation : SharedStorageOperation {
+      Promise<long> run(object data,
+                        FrozenArray<SharedStorageUrlWithMetadata> urls);
+    };
+
+    dictionary SharedStorageUrlWithMetadata {
+      USVString url;
+      object reportingMetadata;
+    };
+  </xmp>
+
+  ## Registering a Worklet ## {#register-worklet}
+  <div algorithm>
+    For the <dfn method for="SharedStorageWorkletGlobalScope">register(|name|, |operationCtor|)</dfn> method, run the following steps [=in parallel=]:
+
+    1. If |name| is missing or empty, throw a {{TypeError}}.
```

Comment:
You need to perform these steps before entering the `[=in parallel=]` block.  You can't throw any exceptions or access javascript state within the parallel code.

So move this up and then have a step that says "Run the following steps [=in parallel=]:", etc.

Reply:
Removed "[=in parallel=]" per your comment below and xyaoinum@'s comment.